base/src/types/mod.rs

```diff
@@ -240,6 +242,21 @@ pub struct TypeVariable {
 #[cfg_attr(feature = "serde_derive", serde(de_parameters = "T"))]
 #[cfg_attr(feature = "serde_derive", serde(serialize_state = "SeSeed"))]
 #[cfg_attr(feature = "serde_derive", serde(bound(serialize = "Id: SerializeState<SeSeed>")))]
+pub struct Skolem<Id> {
+    #[cfg_attr(feature = "serde_derive", serde(state))] pub name: Id,
+    pub id: u32,
+    #[cfg_attr(feature = "serde_derive", serde(state))] pub kind: ArcKind,
+}
+
+#[derive(Clone, Debug, Eq, PartialEq, Hash)]
+#[cfg_attr(feature = "serde_derive", derive(DeserializeState, SerializeState))]
+#[cfg_attr(feature = "serde_derive", serde(deserialize_state = "Seed<Id, T>"))]
+#[cfg_attr(feature = "serde_derive",
+           serde(bound(deserialize = "
+           Id: DeserializeState<'de, Seed<Id, T>> + Clone + ::std::any::Any")))]
+#[cfg_attr(feature = "serde_derive", serde(de_parameters = "T"))]
+#[cfg_attr(feature = "serde_derive", serde(serialize_state = "SeSeed"))]
+#[cfg_attr(feature = "serde_derive", serde(bound(serialize = "Id: SerializeState<SeSeed>")))]
```

Comment:
Lol, these annotations are getting kinda hilarious.

Reply:
Just now "getting hilarious" :) ? They have always been hilarious 🤡 

Its mostly the `cfg_attr` part that makes it messy though. Yesterday I shortened it a bit in https://github.com/gluon-lang/gluon/pull/373 and thought that maybe rustc could automatically treat custom attributes as wrapped in `cfg_attr` if the dependency is optional...

```rust
#[serde(de_parameters = "T")]
// => desugared to
#[cfg_attr(feature = "serde", serde(de_parameters = "T"))]
```

Might be worth floating the idea in internals though I think attributes are in a slightly hacky state right now which may make it hard to get implemented.